fairscale/optim/oss.py

```diff
@@ -340,18 +343,18 @@ def consolidate_state_dict(self, recipient_rank: int = 0) -> None:
                 state_to_share = (
                     self.optim.state_dict()
                     if should_send_state
-                    else torch.tensor([0], dtype=torch.uint8, device=self._default_device)
+                    else torch.tensor([0], dtype=torch.uint8, device=dist_device)
```

Comment:
This seems wasteful. Why not skip the broadcast in this case instead of sending a zero? In the `else` below you could check if `rank != recipient`.